app/components/support_interface/reasons_for_rejection_dashboard_component.rb

```diff
@@ -0,0 +1,45 @@
+module SupportInterface
+  class ReasonsForRejectionDashboardComponent < ViewComponent::Base
+    def initialize(rejection_reasons)
+      @rejection_reasons = rejection_reasons
+    end
+
+    def current_month_rejection_count(reason)
+      values = @rejection_reasons.find { |h| h['key'] == reason && h['time_period'] == 'this_month' }
+      return 0 if values.nil?
+
+      values['count']
+    end
+
+    def total_rejection_count(reason)
+      current_month_rejection_count(reason) + previous_rejection_count(reason)
+    end
+
+    def percentage_rejected_for_reason(reason)
+      count = current_month_rejection_count(reason) + previous_rejection_count(reason)
+      total = @rejection_reasons.inject(0) { |sum, hash| sum + hash['count'] }
+      formatted_percentage(count, total)
+    end
+
+  private
+
+    def previous_rejection_count(reason)
+      values = @rejection_reasons.find { |h| h['key'] == reason && h['time_period'] == 'before_this_month' }
+      return 0 if values.nil?
+
+      values['count']
+    end
+
+    def formatted_percentage(count, total)
+      percentage = count.percent_of(total)
+      precision = (percentage % 1).zero? ? 0 : 2
+      number_to_percentage(percentage, precision: precision)
+    end
+  end
+end
+
+class Numeric
```

Comment:
We probably shouldn't be monkey-patching standard lib classes (even though Rails does it!) Maybe this would be better as a helper method if there's nothing else built-in to do the job. (Helpers are accessible from components if you mix in `ViewHelper` I think).

Reply:
Thanks Steve refactored accordingly